pkg/process/checks/net.go

```diff
@@ -130,48 +131,70 @@ func (c *ConnectionsCheck) enrichConnections(conns []*model.Connection) []*model
+		if _, ok := tel[telemetryTypeName]; ok {
+			cct[metricName] = tel[telemetryTypeName]
+		}
+	}
+
+	metricNameForMonotonicTelemetryType := map[string]string{
+		"kprobes_triggered":           string(network.MonotonicKprobesTriggered),
+		"kprobes_missed":              string(network.MonotonicKprobesMissed),
+		"conntrack_registers":         string(network.MonotonicConntrackRegisters),
+		"conntrack_registers_dropped": string(network.MonotonicConntrackRegistersDropped),
+		"dns_packets_processed":       string(network.MonotonicDNSPacketsProcessed),
+		"conns_closed":                string(network.MonotonicConnsClosed),
+		"udp_sends_processed":         string(network.MonotonicUDPSendsProcessed),
+		"udp_sends_missed":            string(network.MonotonicUDPSendsMissed),
+		"dns_packets_dropped":         string(network.MonotonicDNSPacketsDropped),
 	}
-	c.saveTelemetry(tel)
+
+	for metricName, telemetryTypeName := range metricNameForMonotonicTelemetryType {
+		if _, ok := tel[telemetryTypeName]; ok {
+			cct[metricName] = tel[telemetryTypeName] - c.lastTelemetry[telemetryTypeName]
+		}
+	}
+
+	c.saveMonotonicTelemetry(tel)
 	return cct
 }
 
-func (c *ConnectionsCheck) saveTelemetry(tel *model.ConnectionsTelemetry) {
+func (c *ConnectionsCheck) saveMonotonicTelemetry(tel map[string]int64) {
 	if tel == nil || c.lastTelemetry == nil {
 		return
 	}
 
-	c.lastTelemetry.KprobesTriggered = tel.MonotonicKprobesTriggered
-	c.lastTelemetry.KprobesMissed = tel.MonotonicKprobesMissed
-	c.lastTelemetry.ConntrackRegisters = tel.MonotonicConntrackRegisters
-	c.lastTelemetry.ConntrackRegistersDropped = tel.MonotonicConntrackRegistersDropped
-	c.lastTelemetry.DnsPacketsProcessed = tel.MonotonicDnsPacketsProcessed
-	c.lastTelemetry.ConnsClosed = tel.MonotonicConnsClosed
-	c.lastTelemetry.UdpSendsProcessed = tel.MonotonicUdpSendsProcessed
-	c.lastTelemetry.UdpSendsMissed = tel.MonotonicUdpSendsMissed
-	c.lastTelemetry.DnsStatsDropped = tel.DnsStatsDropped
+	c.lastTelemetry[string(network.MonotonicKprobesTriggered)] = tel[string(network.MonotonicKprobesTriggered)]
+	c.lastTelemetry[string(network.MonotonicKprobesMissed)] = tel[string(network.MonotonicKprobesMissed)]
+	c.lastTelemetry[string(network.MonotonicConntrackRegisters)] = tel[string(network.MonotonicConntrackRegisters)]
+	c.lastTelemetry[string(network.MonotonicConntrackRegistersDropped)] = tel[string(network.MonotonicConntrackRegistersDropped)]
+	c.lastTelemetry[string(network.MonotonicDNSPacketsProcessed)] = tel[string(network.MonotonicDNSPacketsProcessed)]
+	c.lastTelemetry[string(network.MonotonicConnsClosed)] = tel[string(network.MonotonicConnsClosed)]
+	c.lastTelemetry[string(network.MonotonicUDPSendsProcessed)] = tel[string(network.MonotonicUDPSendsProcessed)]
+	c.lastTelemetry[string(network.MonotonicUDPSendsMissed)] = tel[string(network.MonotonicUDPSendsMissed)]
+	c.lastTelemetry[string(network.MonotonicDNSPacketsDropped)] = tel[string(network.MonotonicDNSPacketsDropped)]
```

Comment:
Do you think it would make sense to have a list of all the monotonic counters defined next to the constants, that this could just loop through?

Reply:
We can definitely do that!